vtree/diff.js

```diff
@@ -343,16 +348,23 @@ function reorder(aChildren, bChildren) {
 // TODO: We really want to make this more efficient
 // Finding the current index of any key in the array could be
 // faster.
-function moveItem(array, toIndex, itemKey) {
+function indexOfKey(array, itemKey) {
```

Comment:
the list mutates during the patch so it has to update after each mutation... which is no more efficient than looking for the item each time we mutate (like we do now).

...and to cut off another thought, `array.indexOf` is likely to perform just the same as what we're doing but only some performance tests can really prove if it's faster.